src/embedded_graphics.rs

```diff
@@ -54,37 +66,47 @@ where
     }
 }
 
-impl<const FB_WIDTH: usize, const FB_HEIGHT: usize> DrawTarget for Framebuffer<{FB_WIDTH}, {FB_HEIGHT}>
+const fn line_bytes(width: u32, bits_per_pixel: usize) -> usize {
+    (width as usize * bits_per_pixel + 7) / 8
+}
+
+impl<const FB_WIDTH: usize, const FB_HEIGHT: usize> DrawTarget
+    for Framebuffer<{ FB_WIDTH }, { FB_HEIGHT }>
 where
     [(); FB_WIDTH * FB_HEIGHT / 8]:,
 {
     type Color = Color;
     type Error = ();
 
-    fn draw_iter<I>(&mut self, pixels: I) -> Result<(), Self::Error>
+    fn draw_iter<I>(&mut self, pixels: I) -> Result<(), ()>
     where
-        I: IntoIterator<Item = embedded_graphics_core::Pixel<Self::Color>>,
+        I: IntoIterator<Item = embedded_graphics_core::Pixel<Color>>,
     {
         for embedded_graphics_core::Pixel(point, color) in pixels {
-            if point.x < 0 || point.x >= FB_WIDTH as i32 || point.y < 0 || point.y >= FB_HEIGHT as i32 {
-                continue;
-            }
+            // Maybe implement it for the enum?
+            let color_value = match color {
+                Color::White => 1,
+                Color::Black => 0,
+            };
+
             let x = point.x as usize;
             let y = point.y as usize;
-            let byte_index = x / 8 + y * FB_WIDTH / 8;
-            let byte = &mut self.framebuffer[byte_index];
-            let bit_index = 7 - x % 8;
 
-            match color {
-                Color::White => {
-                    *byte |= 0b1 << bit_index;
-                }
-                Color::Black => {
-                    *byte &= !(0b1 << bit_index);
-                }
+            if x >= FB_WIDTH || y >= FB_HEIGHT {
+                continue;
             }
-        }
 
+            let index = (y * FB_WIDTH + x) / 8;
+            let bit = 7 - (x % 8);
```

Comment:
this calculation needs to be moved to write_image (or whatever lower level function it calls) so that embedded graphics is not required to do partial updates

but I am working on this now

Reply:
nvm, I just remembered it can't be done this way because the driver doesn't have a way to "remember" the values of the bits that are on the margin of the aligned area. I think I was originally going to do it pretty much the way that you have, the FB is really the place for this logic after all